File: php/features.php
<?php
declare(strict_types=1);

// BizFactory SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class BizFactoryFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new BizFactoryBaseFeature();
            case "test":
                return new BizFactoryTestFeature();
            default:
                return new BizFactoryBaseFeature();
        }
    }
}

File: py/features.py
# BizFactory SDK feature factory

from feature.base_feature import BizFactoryBaseFeature
from feature.test_feature import BizFactoryTestFeature


def _make_feature(name):
    features = {
        "base": lambda: BizFactoryBaseFeature(),
        "test": lambda: BizFactoryTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
